clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -258,7 +258,10 @@ public void poll(long now) {
         }
 
         pollHeartbeat(now);
-        maybeAutoCommitOffsetsAsync(now);
+        if (subscriptions.hasNoSubscription())
+            subscriptions.clearAssignment();
```

Comment:
I'm wondering if it makes sense to move this to `onJoinPrepare` (after invocation of `onPartitionsRevoked`) since that is the last step before rejoining the group?